docusaurus/docs/learn/quickstarts/network/miniziti.bash

```diff
@@ -169,36 +189,60 @@ installHosts() {
     fi
 }
 
-logInfo() {
-    (( $# )) || {
-        logError "logInfo() takes 1 or more args"
+getAdminSecret() {
+    kubectl_wrapper get secrets "ziti-controller-admin-secret" \
+        --namespace "${ZITI_NAMESPACE}" \
+        --output go-template='{{index .data "admin-password" | base64decode }}'
+}
+
+showAdminCreds() {
+    logInfo "The password for 'admin' is: '$(getAdminSecret)'"
+}
+
+miniziti_login() {
+    getAdminSecret | xargs ziti edge login  \
+            --cli-identity "$MINIKUBE_PROFILE" \
+            --username "admin" \
+            --password
+}
+
+logger() {
+    local caller="${FUNCNAME[1]}"
+    if (( $# < 1 )); then
+        echo "ERROR: $caller() takes 1 or more args" >&2
         return 1
-    }
-    echo "INFO: $*"
+    fi
+
+    caller_level="${caller##log}"
+    if (( MINIZITI_DEBUG )); then
+        echo -e "${caller_level^^} ${FUNCNAME[2]}:${BASH_LINENO[1]}: $*"
+    else
+        echo -e "${caller_level^^} $*"
+    fi
+}
+
+logInfo() {
+    logger "$*"
 }
 
 logWarn() {
-    (( $# )) || {
-        logError "logWarn() takes 1 or more args"
-        return 1
-    }
-    echo "WARN: $*" >&2
+    logger "$*" >&2
 }
 
 logError() {
-    (( $# )) || {
-        echo "ERROR: logError() takes 1 or more args" >&2
-        return 1
-    }
-    echo "ERROR: $*" >&2
+    logger "$*" >&2
 }
 
 logDebug() {
-    (( $# )) || {
-        logError "logDebug() takes 1 or more args"
-        return 1
-    }
-    echo "DEBUG: $*" >&3
+    logger "$*" >&3
+}
+
+ziti_wrapper() {
```

Comment:
Possibly a future feature for another PR, but it'd be convenient if this function performed a login when necessary. That way, it's never necessary to explicitly to know about or think about running the login command by itself, just the CLI wrapper. The CLI JSON file has a `loginTime` property for the profile, and it might be possible to discover when the token expires. For example,

```bash
# pseudo bash
ziti_wrapper(){
if login profile exists; then
    if loginTime > expiry; then
        miniziti_login()
    fi
else
    miniziti_login()
fi
ziti "$@" --cli-identity "$MINIKUBE_PROFILE"
}
```

Reply:
Agreed, and on several lines, I was tempted to add in the `jq` dependency. Any objection? We take out odd `awk` statements that way

Reply:
I don't want to deal with JSON outside of `jq` if I can help it, so can we leave this request for a future feature?

Reply:
Good call on that.